src/main/java/service/operation/AddOperation.java

```diff
@@ -0,0 +1,26 @@
+package service.operation;
+
+import dao.FruitsDao;
+import exception.InvalidInputException;
+import model.Fruit;
+import model.FruitRecordDto;
+
+public class AddOperation implements Operation {
+    private FruitsDao fruitsDao;
+
+    public AddOperation(FruitsDao fruitsDao) {
+        this.fruitsDao = fruitsDao;
+    }
+
+    @Override
+    public boolean apply(FruitRecordDto fruitRecordDto) {
+        Fruit fruit = fruitsDao.get(fruitRecordDto.getFruitType());
+        int newAmount = fruitsDao.getAmount(fruit) + fruitRecordDto.getAmount();
+        try {
```

Comment:
what happens if you return false ? 
If you process the report anyway it's better not to do this since some records will be ignored but user won't now that